### PR TITLE
Test static checks

### PR DESCRIPTION
**Pull request by yiqian1**
-  Add a new "Static Test" Jenkins stage.
 -  Set the path to the clang-format executable in the "Static Test" sh step.
 -  Invoke the static tests after cmake build to make generated .inc files accessible by clang-tidy.

### CODE REVIEW

File: mlir/utils/jenkins/Jenkinsfile.downstream [hunk 1]
```diff
@@ -21,14 +21,15 @@ pipeline {
...
+                    pip3 install pathspec unidiff
                 '''
             }
         }
-        stage('Test') {
+        stage('Build and Test') {
             steps {
                 checkout scm
-                sh '''
+                sh '''
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile.downstream]: Please remove the trailing whitespace.

File: mlir/utils/jenkins/Jenkinsfile.downstream [hunk 2]
```diff
@@ -47,9 +48,27 @@ pipeline {
...
+                    fi
+
+		    # Set the path to the clang-format executable 
+                    PATH=$PATH:/opt/rocm/llvm/bin/
+
+		    python3 ./mlir/utils/jenkins/static-checks/premerge-checks.py
+
+                """ 
+            }
+	}
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile.downstream]: Please make the tabspace to be 4 spaces to be consistent with rest of the file.

**Comment by jerryyin** [comment]
Could you point out what did you change to fix #15 issues?

**Comment by yiqian1** [comment]
Where should I include how I fixed previous errors? In the commit message?

**Comment by jerryyin** [comment]
@yiqian1 Since you don't have a dedicated commit that does the fix, commit message won't be a good place for it. The best place should be in this PR's top message body.

**Review by jerryyin — APPROVED** [review]

**Pull request merged**